packages/lesswrong/server/authentication/auth0.ts

```diff
@@ -0,0 +1,55 @@
+import { DatabaseServerSetting } from "../databaseSettings";
+import { ManagementClient } from "auth0";
+import Profile from "passport-auth0/lib/Profile";
+
+type Auth0Settings = {
+  appId: string;
+  secret: string;
+  domain: string;
+  originalDomain: string;
+}
+
+/**
+ * Auth0 Management API Client
+ * For this to work, we must authorize the forum application in Auth0:
+ * Applications -> APIs -> Auth0 Management API -> Machine to Machine Applications
+ */
+const auth0Client = new class {
```

Comment:
Is there a reason for this to be an anonymous class instance, instead of a regular class?  i.e.

```
class Auth0Client {
  ...
}

const auth0Client = new Auth0Client();
```

Also, long-term, worth considering whether [missing config/secrets here should cause server start to fail](https://lworg.slack.com/archives/C75BWSNBH/p1658975737892079)

Reply:
There's no reason we couldn't do it like that, but we don't ever want more than one instance of this so I'm not sure there's any reason _to_ do it like that. I've just given the class a name to make sure we don't get any annoying "anonymous class" messages in any error messages or stack traces.

As it's written at the moment, missing environment variables won't cause any issues until you actually try to make a request to the Auth0 API.